flink-connector-opensearch/src/main/java/org/apache/flink/connector/opensearch/sink/OpensearchAsyncSinkBuilder.java

```diff
@@ -0,0 +1,246 @@
+        this.connectionRequestTimeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the timeout for establishing a connection of the Opensearch cluster.
+     *
+     * @param timeout timeout for the connection (in milliseconds)
+     * @return this builder
+     */
+    public OpensearchAsyncSinkBuilder<InputT> setConnectionTimeout(int timeout) {
+        checkArgument(timeout >= 0, "Connection timeout must be larger than or equal to 0.");
+        this.connectionTimeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the timeout for establishing a connection of the Opensearch cluster.
+     *
+     * @param timeout timeout for the connection (in milliseconds)
+     * @param timeUnit timeout time unit
+     * @return this builder
+     */
+    public OpensearchAsyncSinkBuilder<InputT> setConnectionTimeout(int timeout, TimeUnit timeUnit) {
+        checkNotNull(timeUnit, "TimeUnit cannot be null.");
+        return setConnectionTimeout((int) timeUnit.toMillis(timeout));
+    }
+
+    /**
+     * Sets the timeout for waiting for data or, put differently, a maximum period inactivity
+     * between two consecutive data packets.
+     *
+     * @param timeout timeout for the socket (in milliseconds)
+     * @return this builder
+     */
+    public OpensearchAsyncSinkBuilder<InputT> setSocketTimeout(int timeout) {
+        checkArgument(timeout >= 0, "Socket timeout must be larger than or equal to 0.");
+        this.socketTimeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the timeout for waiting for data or, put differently, a maximum period inactivity
+     * between two consecutive data packets.
+     *
+     * @param timeout timeout for the socket
+     * @param timeUnit timeout time unit
+     * @return this builder
+     */
+    public OpensearchAsyncSinkBuilder<InputT> setSocketTimeout(int timeout, TimeUnit timeUnit) {
+        checkNotNull(timeUnit, "TimeUnit cannot be null.");
+        return setSocketTimeout((int) timeUnit.toMillis(timeout));
+    }
+
+    /**
+     * Allows to bypass the certificates chain validation and connect to insecure network endpoints
+     * (for example, servers which use self-signed certificates).
+     *
+     * @param allowInsecure allow or not to insecure network endpoints
+     * @return this builder
+     */
+    public OpensearchAsyncSinkBuilder<InputT> setAllowInsecure(boolean allowInsecure) {
+        this.allowInsecure = allowInsecure;
+        return this;
+    }
+
+    @Override
+    public OpensearchAsyncSink<InputT> build() {
+        return new OpensearchAsyncSink<InputT>(
+                nonNullOrDefault(
+                        getMaxBatchSize(),
+                        1000), /* OpensearchConnectorOptions.BULK_FLUSH_MAX_ACTIONS_OPTION */
+                nonNullOrDefault(
+                        getMaxInFlightRequests(), 1), /* BulkProcessor::concurrentRequests */
+                nonNullOrDefault(getMaxBufferedRequests(), 10000),
```

Comment:
Can we also promote the other magic numbers to constants? `10000` and `2 * 1024 * 1024`